vm/src/vm/runners/cairo_pie.rs

```diff
@@ -41,6 +41,15 @@ pub struct CairoPieMemory(
     pub  Vec<((usize, usize), MaybeRelocatable)>,
 );
 
+impl CairoPieMemory {
+    pub fn as_hashmap(&self) -> HashMap<&(usize, usize), &MaybeRelocatable> {
```

Comment:
what do you think about manually implementing `PartialEq` for `CairoPieMemory` and putting the logic to convert to hashmap there? So there is a single way of checking whether 2 CairoPieMemories are equal or not. Having 2 ways can lead to confusion.

Reply:
Sounds good!